docs: add `x/subspaces` spec (#910)
desmos-labs/desmos: x/subspaces/spec/03-state.md

## Next Section ID
The Next Section ID is stored on the chain using its associated subspace ID as key.

* Next Section ID: `0x06 | Subspace ID | -> bytes(NextSectionID)`

## Section
The Section is stored using both the Subspace ID and its ID as keys. This make it easier to query:
- All the subspace related sections;
- A specific section inside a given subspace.

* Section: `0x07 | Subspace ID | Section ID | -> ProtocolBuffer(Section)`

## Next Group ID
The Next Group ID is stored on the chain using its associated subspace ID as key:

* Next Group ID: `0x02 | Subspace ID | -> bytes(NextUserGroupID)`

## User Group
The user group is stored on the chain with a combination of subspace, section and user group IDs as key. This make it easier to query:
- All the user groups of a subspace;
- All the user groups of a section;
- A specific user group.

* User Group: `0x03 | Subspace ID | Section ID | User Group ID -> ProtocolBuffer(UserGroup)`

## User Group Member
The user group member is stored on the chain with a combination of subspace and user group IDs as key.

* User Group Member: `0x04 | Subspace ID | User Group ID | Address | -> 0x01`

## User Permission

> The user permission is stored on the chain with a combination of Subspace ID, Section ID and user address as key. This make it easier to query:
> - All the User Permissions for a subspace;
> - All the User Permissions for a section;
> - All the User Permissions for an address.

Reviewer: ```suggestion
A user permission is stored on the chain with a combination of subspace id, section id and user address as key. This make it easy to query:
- all the user permissions set within a subspace;
- all the user permissions set within a a section;
- all the permissions set to an address.
```
